### PR TITLE
lesson-1

### PR DESCRIPTION
**Pull request by ThinhNgVhust**

### CODE REVIEW

**Comment by ThinhNgVhust** [comment]
add reverse linked list problem set

**Comment by quanghoang33** [comment]
lgtm

**Comment by ThinhNgVhust** [comment]
Lesson3

File: everything/GC.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Gabage Collection
+If you are not reachable from root set(set of variables from stack), I delete you to make the earth cleaner!!!  
+![image](imgs/Animation_of_the_Naive_Mark_and_Sweep_Garbage_Collector_Algorithm.gif)  
+Naive mark-and-sweep in action on a **heap** containing eight objects. Arrows represent object references. Circles represent the objects themselves. Objects #1, #2, #3, #4, and #6 are strongly referenced from the root set. On the other hand, objects #5, #7, and #8 are not strongly referenced either directly or indirectly from the root set; therefore, they are garbage.
+
+## What is it?
```
Comment (quanghoang33) [inline everything/GC.md:6]: consider add more deep detail, how it work . ref: https://www.oracle.com/webfolder/technetwork/tutorials/obe/java/gc01/index.html

Comment (quanghoang33) [reply everything/GC.md:6]: it is a big PR already so I will merge it first.

**Review by quanghoang33 — COMMENTED** [review]
algo part lgtm

**Pull request merged**